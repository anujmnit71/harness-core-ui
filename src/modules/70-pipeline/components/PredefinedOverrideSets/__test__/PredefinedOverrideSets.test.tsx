/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { PredefinedOverrideSets } from '@pipeline/components/PredefinedOverrideSets/PredefinedOverrideSets'
import type { DeploymentStageElementConfig, StageElementWrapper } from '@pipeline/utils/pipelineTypes'
jest.mock('framework/strings', () => ({
  useStrings: () => ({
    getString: (key: string) => key
  })
}))
const currentStageMock: StageElementWrapper<DeploymentStageElementConfig> = {
  stage: {
    spec: {
      execution: {
        rollbackSteps: undefined,
        steps: []
      },
      infrastructure: {
        allowSimultaneousDeployments: false,
        environment: undefined,
        environmentRef: 'environmentRef',
        infrastructureDefinition: undefined,
        infrastructureKey: 'infrastructureKey',
        useFromStage: undefined
      },
      serviceConfig: {
        service: undefined,
        serviceDefinition: undefined,
        serviceRef: 'serviceRef',
        stageOverrides: {
          artifacts: undefined,
          manifests: undefined,
          variables: undefined,
          useArtifactOverrideSets: ['artifact1', 'artifact2'],
          useManifestOverrideSets: ['manifest1'],
          useVariableOverrideSets: ['variable1', 'variable2']
        },
        useFromStage: undefined
      }
    },
    identifier: 'identifier',
    name: 'name'
  }
}
describe('Test ase for OverrideSetsInputSelector', () => {
  test('Snapshot test OverrideSetsInputSelector - ARTIFACT', () => {
    const artifactContext = 'ARTIFACT'
    const { container } = render(<PredefinedOverrideSets currentStage={currentStageMock} context={artifactContext} />)
    expect(container).toMatchSnapshot('Artifact Context')
  })
  test('Snapshot test OverrideSetsInputSelector - MANIFEST', () => {
    const artifactContext = 'MANIFEST'
    const { container } = render(<PredefinedOverrideSets currentStage={currentStageMock} context={artifactContext} />)
    expect(container).toMatchSnapshot('MANIFEST Context')
  })
  test('Snapshot test OverrideSetsInputSelector - VARIABLE', () => {
    const artifactContext = 'VARIABLES'
    const { container } = render(<PredefinedOverrideSets currentStage={currentStageMock} context={artifactContext} />)
    expect(container).toMatchSnapshot('VARIABLES Context')
  })
})
