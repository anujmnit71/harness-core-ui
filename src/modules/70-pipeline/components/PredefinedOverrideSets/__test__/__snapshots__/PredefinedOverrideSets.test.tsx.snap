// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ase for OverrideSetsInputSelector Snapshot test OverrideSetsInputSelector - ARTIFACT: Artifact Context 1`] = `
<div>
  <div
    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex"
    style="align-items: center; justify-content: end;"
  >
    <p
      class="StyledProps--font StyledProps--main"
    >
      Use Predefined Override Sets
    </p>
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <button
          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
          style="width: 350px;"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            <span
              class="Tag--main"
            >
              <span
                class="bp3-tag bp3-minimal tag"
              >
                <span
                  class="bp3-text-overflow-ellipsis bp3-fill"
                >
                  artifact1
                </span>
              </span>
            </span>
            <span
              class="Tag--main"
            >
              <span
                class="bp3-tag bp3-minimal tag"
              >
                <span
                  class="bp3-text-overflow-ellipsis bp3-fill"
                >
                  artifact2
                </span>
              </span>
            </span>
          </span>
          <span
            class="bp3-icon bp3-icon-caret-down StyledProps--main"
            icon="caret-down"
          >
            <svg
              data-icon="caret-down"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-down
              </desc>
              <path
                d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;

exports[`Test ase for OverrideSetsInputSelector Snapshot test OverrideSetsInputSelector - MANIFEST: MANIFEST Context 1`] = `
<div>
  <div
    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex"
    style="align-items: center; justify-content: end;"
  >
    <p
      class="StyledProps--font StyledProps--main"
    >
      Use Predefined Override Sets
    </p>
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <button
          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
          style="width: 350px;"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            <div
              class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
            >
              <span>
                manifest1
              </span>
              <button
                class="clearButton"
              >
                <span
                  class="bp3-icon bp3-icon-cross StyledProps--main"
                  icon="cross"
                >
                  <svg
                    data-icon="cross"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      cross
                    </desc>
                    <path
                      d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </span>
          <span
            class="bp3-icon bp3-icon-caret-down StyledProps--main"
            icon="caret-down"
          >
            <svg
              data-icon="caret-down"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-down
              </desc>
              <path
                d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;

exports[`Test ase for OverrideSetsInputSelector Snapshot test OverrideSetsInputSelector - VARIABLE: VARIABLES Context 1`] = `
<div>
  <div
    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--flex"
    style="align-items: center; justify-content: end;"
  >
    <p
      class="StyledProps--font StyledProps--main"
    >
      Use Predefined Override Sets
    </p>
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <button
          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
          style="width: 350px;"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            <span
              class="Tag--main"
            >
              <span
                class="bp3-tag bp3-minimal tag"
              >
                <span
                  class="bp3-text-overflow-ellipsis bp3-fill"
                >
                  variable1
                </span>
              </span>
            </span>
            <span
              class="Tag--main"
            >
              <span
                class="bp3-tag bp3-minimal tag"
              >
                <span
                  class="bp3-text-overflow-ellipsis bp3-fill"
                >
                  variable2
                </span>
              </span>
            </span>
          </span>
          <span
            class="bp3-icon bp3-icon-caret-down StyledProps--main"
            icon="caret-down"
          >
            <svg
              data-icon="caret-down"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-down
              </desc>
              <path
                d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;
